app/Http/Controllers/ProductController.php

```diff
@@ -36,13 +37,23 @@ public function create()
         return view('addproduct');
     }
 
-    public function store(){
+    public function store(Request $request){
+
         $product = new Product;
+
+        $request->validate([
+            'productName' => 'required|unique:products|string',
+            'productBrand' => 'nullable|string',
+            'productPictureUrl' => 'nullable|active_url',
+            'added_by' => 'integer',
```

Comment:
Not used as you get the id from the authenticated user, right? (which is a good thing)

Reply:
When you add a product he fills the added_by in with the id of the user that is logged in at that moment. Then he goes with the rest of the data to the validator and than he adds it to the database

Reply:
I don't know if that answers your answer somewhere